### PR TITLE
Feature - docs: expand troubleshooting page and add installation page

### LINKED ISSUE #1091: Installation docs are missing
Hello, I can't find any docs for the installation. Is this because of the upgrade to version 4?

http://www.sublimelinter.com/en/latest/installation.html

There are also no docs in the repository here for the installation.

### PR DESCRIPTION
**Pull request by nvtkaszpir**
Should theoretically reduce number of issues reported on GitHub ;)
Fixes #1091.

### CODE REVIEW

**Comment by braver** [comment]
Well, at least it gives us a page to point to ;)

File: docs/troubleshooting.rst [hunk 1]
```diff
@@ -23,16 +23,38 @@ Here are the most common reasons why a linter does not work:
...
 
-- The linter binary is not installed.
+- The linter binary or it dependencies are not installed.
   Be sure to install the linter as documented in the linter plugin’s README.
 
 - The linter binary is installed,
   but its path is not available to SublimeLinter.
   Follow the steps in :ref:`debugging-path-problems` below.
 
+Less common reasons:
```
Comment (braver) [inline docs/troubleshooting.rst]: I wouldn't want to add a section like this. For the really uncommon reasons, either a user knows how to debug, or he asks us questions. I'd rather resolve them in the specific issue and perhaps search through the issues when something familiar comes up, than try to maintain a document about it. 

E.g. All plugins that aren't compatible already have issues open for this. Some simply aren't maintained.

If there are conflicts like with gutter icons or popups, either a user already knows basic troubleshooting tactics and figures this out without the docs, or he's simply not going to troubleshoot (or do it wrong).

Comment (nvtkaszpir) [reply docs/troubleshooting.rst]: OK, I get the idea.
I suggest replacing it with just ' Anything else should be reported as GitHub issue.'?

**Comment by nvtkaszpir** [comment]
One more thing, maybe it is worth to update https://github.com/SublimeLinter/SublimeLinter/blob/master/messages/install.txt with the url at the end? (SublimeLinter instead of SublimeLinter3)

File: docs/troubleshooting.rst (hunk 1 above)
Comment (braver) [reply docs/troubleshooting.rst]: Yeah, that’s a good idea.

**Comment by braver** [comment]
Good catch, that should be without the 3

File: docs/troubleshooting.rst [hunk 2]
```diff
@@ -23,16 +23,19 @@ Here are the most common reasons why a linter does not work:
   to a known syntax (e.g. ``"html"``). The detected syntax is printed to the
   console in debug mode.
   Also note that plugins should move to using the selector setting
-  instead of the old syntaxes attribute. You can use the "selector" linter
+  instead of the old syntaxes attribute. You can use the :ref:`selector` linter
   setting right now instead of the "syntax_map".
 
-- The linter binary is not installed.
+- The linter binary or it dependencies are not installed.
```
Comment (FichteFoll) [inline docs/troubleshooting.rst]: Its

File: docs/troubleshooting.rst [hunk 3]
```diff
@@ -23,16 +23,19 @@ Here are the most common reasons why a linter does not work:
...
   Be sure to install the linter as documented in the linter plugin’s README.
 
 - The linter binary is installed,
   but its path is not available to SublimeLinter.
   Follow the steps in :ref:`debugging-path-problems` below.
 
+Anythng else should be searched across GitHub issues and if not found then
+reported.
```
Comment (FichteFoll) [inline docs/troubleshooting.rst]: Maybe also add a sentence about closed issues? I think many users only consider open ones.

File: docs/troubleshooting.rst [hunk 4]
```diff
@@ -23,16 +23,19 @@ Here are the most common reasons why a linter does not work:
...
 
-- The linter binary is not installed.
+- The linter binary or it dependencies are not installed.
   Be sure to install the linter as documented in the linter plugin’s README.
 
 - The linter binary is installed,
   but its path is not available to SublimeLinter.
   Follow the steps in :ref:`debugging-path-problems` below.
 
+Anythng else should be searched across GitHub issues and if not found then
```
Comment (FichteFoll) [inline docs/troubleshooting.rst]: Anything

File: docs/installation.rst [hunk 5]
```diff
@@ -0,0 +1,11 @@
+Installation
+==================
+
+SublimeLinter and corresponding linter plugins should be installed using `PackageControl <https://packagecontrol.io/installation>`_.
+
+First install PackageControl and then see its `usage <https://packagecontrol.io/docs/usage>`_,
+which explains how to install plugins to SublimeText.
```
Comment (FichteFoll) [inline docs/installation.rst]: The correct term is "packages" here.

File: docs/installation.rst [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+Notice, that it may be required to restart SublimeText after plugin installation.
```
Comment (FichteFoll) [inline docs/installation.rst]: The comma is wrong. Or well, I certainly would not write it.

File: docs/installation.rst [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+Make sure to read `SublimeLinter messages <https://github.com/SublimeLinter/SublimeLinter/tree/master/messages>`_.
```
Comment (FichteFoll) [inline docs/installation.rst]: This reference seems odd. Messages are shown on updates, so users should rather read the messages that apply to them in their editor when they open.

**Comment by FichteFoll** [comment]
Aw, I was too slow.

**Comment by braver** [comment]
I’m just glad to have it written and merged. I usually re-read and correct docs afterwards, just before releases. I don’t like doing the language corrections and typos via code review, instead I edit afterwards.

File: docs/troubleshooting.rst (hunk 2 above)
Comment (nvtkaszpir) [reply docs/troubleshooting.rst]: fixed

File: docs/troubleshooting.rst (hunk 3 above)
Comment (nvtkaszpir) [reply docs/troubleshooting.rst]: Good idea.

File: docs/troubleshooting.rst (hunk 4 above)
Comment (nvtkaszpir) [reply docs/troubleshooting.rst]: fixed.

File: docs/installation.rst (hunk 5 above)
Comment (nvtkaszpir) [reply docs/installation.rst]: fixed

File: docs/installation.rst (hunk 6 above)
Comment (nvtkaszpir) [reply docs/installation.rst]: fixed.

File: docs/installation.rst (hunk 7 above)
Comment (nvtkaszpir) [reply docs/installation.rst]: Yes, but looking at the amount of issues on GitHub it seems people are not reading those.
Also AFAIR it shows only last change, not sure if it shows all messages between current and freshly installed versions....

**Comment by nvtkaszpir** [comment]
Fixes will be in new PR.

**Pull request merged**